libact/query_strategies/tests/test_realdata.py

```diff
@@ -0,0 +1,112 @@
+
+    def test_RandomSampling(self):
+        random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:5], [None]*(len(self.y)-5)]))
+        qs = RandomSampling(trn_ds)
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([141, 37, 129, 15, 151, 149, 237, 17, 146, 91]))
+
+    def test_HintSVM(self):
+        np.random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:5], [None]*(len(self.y)-5)]))
+        qs = HintSVM(trn_ds)
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([24, 235, 228, 209, 18, 143, 119, 90, 149, 207]))
+
+    def test_QueryByCommittee(self):
+        #import ipdb; ipdb.set_trace()
+        random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:10], [None]*(len(self.y)-10)]))
+        qs = QueryByCommittee(trn_ds,
+                              models=[LogisticRegression(C=1.0),
+                                      LogisticRegression(C=0.01),
+                                      LogisticRegression(C=100)])
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([ 11, 207, 101, 30, 116, 108, 83, 172, 211, 42]))
+
+    def test_UcertaintySamplingLc(self):
+        random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:10], [None]*(len(self.y)-10)]))
+        qs = UncertaintySampling(trn_ds, method='lc',
+                                 model=LogisticRegression())
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([145, 66, 82, 37, 194, 60, 191, 211, 245, 131]))
+
+    def test_UcertaintySamplingSm(self):
+        random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:10], [None]*(len(self.y)-10)]))
+        qs = UncertaintySampling(trn_ds, method='sm',
+                                 model=LogisticRegression())
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([145, 66, 82, 37, 194, 60, 191, 211, 245, 131]))
+
+    def test_ActiveLearningByLearning(self):
+        np.random.seed(1126)
+        trn_ds = Dataset(self.X,
+                         np.concatenate([self.y[:10], [None]*(len(self.y)-10)]))
+        qs = ActiveLearningByLearning(trn_ds, T=self.quota,
+                query_strategies=[
+                    UncertaintySampling(trn_ds, model=LogisticRegression()),
+                    HintSVM(trn_ds)],
+                model=LogisticRegression()
+            )
+        qseq = run_qs(trn_ds, qs, self.y, self.quota)
+        assert_array_equal(
+            qseq, np.array([103, 220, 118,  75, 176,  50, 247, 199,  46,  55]))
+
+
```

Comment:
Do we want to add test_variance_reduction in libact/query_strategies/tests/test_realdata.py?